keiko-dynomite-spring/src/main/kotlin/com/netflix/spinnaker/config/DynomiteQueueConfiguration.kt

```diff
@@ -0,0 +1,137 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.config
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.netflix.discovery.DiscoveryClient
+import com.netflix.dyno.connectionpool.Host
+import com.netflix.dyno.connectionpool.HostSupplier
+import com.netflix.dyno.connectionpool.TokenMapSupplier
+import com.netflix.dyno.connectionpool.impl.ConnectionPoolConfigurationImpl
+import com.netflix.dyno.connectionpool.impl.lb.HostToken
+import com.netflix.dyno.jedis.DynoJedisClient
+import com.netflix.dyno.queues.redis.RedisQueues
+import com.netflix.dyno.queues.redis.SingleShardSupplier
+import com.netflix.spinnaker.q.DeadMessageCallback
+import com.netflix.spinnaker.q.dynomite.DynomiteDeadMessageHandler
+import com.netflix.spinnaker.q.dynomite.DynomiteQueue
+import com.netflix.spinnaker.q.metrics.EventPublisher
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import java.time.Clock
+import java.time.Duration
+import java.util.*
+
+@Configuration
+@ConditionalOnProperty("keiko.queue.dynomite.enabled")
+@EnableConfigurationProperties(DynomiteQueueConfiguration::class)
+open class DynomiteQueueConfiguration {
+
+  @Bean
+  @ConfigurationProperties("keiko.queue.dynomite.connectionPool")
+  open fun connectionPoolConfigurationImpl(dynomiteConfigurationProperties: DynomiteConfigurationProperties): ConnectionPoolConfigurationImpl {
+    return ConnectionPoolConfigurationImpl(dynomiteConfigurationProperties.applicationName)
+  }
+
+  @Bean(destroyMethod = "stopClient")
+  open fun dynoJedisClient(dynomiteConfigurationProperties: DynomiteConfigurationProperties,
+                           connectionPoolConfiguration: ConnectionPoolConfigurationImpl,
+                           discoveryClient: Optional<DiscoveryClient>): DynoJedisClient
+    = createDynoJedisClient(dynomiteConfigurationProperties, connectionPoolConfiguration, discoveryClient)
+
+  @Bean open fun redisQueues(dynomiteConfigurationProperties: DynomiteConfigurationProperties,
+                           dynoJedisClient: DynoJedisClient): RedisQueues {
+    return RedisQueues(
+      dynoJedisClient,
+      dynoJedisClient,
+      dynomiteConfigurationProperties.keyPrefix,
+      // TODO rz - add support for DynoShardSupplier
```

Comment:
As I understand it, the shard supplier will increase throughput - not something we need as of yet.